### DISCUSSION TITLE
how to tell when client is connected if RetryOnFailedConnect=true

### DISCUSSION THREAD

**Discussion by cbrake** [nats-io/nats.go#1122] [Q&A] [answered]
I'm using the client with RetryOnFailedConnect=true. The nats.Connect() always returns true in this case. 
Is there any way to tell if the initial connection succeeded. There is SetReconnectHandler() which tells me if a reconnect worked, but I would also like to know if the initial connection succeeded so that I know if I can start sending messages.

I could wait a bit and assume if I don't get any CustomReconnectDelay() calls, then I'm probably connected.

Likely missing something here, so appreciate any ideas. Thanks!

**Comment by derekcollison** [comment, marked as answer]
`nc.Status()` I think..

**Reply by cbrake** [reply]
Thanks, that works!

It may make sense at some point to consider adding a ConnectHandler (similar to Reconnect, Disconnect, and Closed handlers).

**Reply by derekcollison** [reply]
I am not positive but I think the reconnect handler might already fire in those instances when the first attempt failed.

**Reply by cbrake** [reply]
Yes, the reconnect handler does fire if the first attempt failed. However, if the first attempt succeeded, I have no way of knowing this other than delaying, and polling nc.Status(). Not a huge deal, but a connected handler would complete the API.

**Reply by derekcollison** [reply]
Fair point! Maybe create issue on Go client if possible. Thanks.

**Reply by cbrake** [reply]
Issue created: https://github.com/nats-io/nats.go/issues/1124 -- thanks!
